iphone/Maps/Classes/CustomViews/NavigationDashboard/MWMNavigationDashboardManager+Entity.mm

```diff
@@ -174,7 +206,11 @@ - (void)updateFollowingInfo:(routing::FollowingInfo const &)info type:(MWMRouter
     entity.speedLimitMps = info.m_speedLimitMps;
 
     entity.estimate = estimate(entity.timeToTarget, entity.targetDistance, entity.targetUnits,
-                               self.etaAttributes, self.etaSecondaryAttributes, NO);
+                               self.etaAttributes, self.etaSecondaryAttributes, NO, showEta);
+    if (type == MWMRouterTypeRuler && [points count] > 2)
+      entity.transitSteps = buildRouteTransitSteps(points);
+    else
+      entity.transitSteps = [[NSArray alloc] init];
```

Comment:
nit: Probably nil is enough here, and check nil instead of empty below

Reply:
Changing to nil requires changing in `TransportTransitStepsCollectionView.swift` field `steps: [MWMRouterTransitStepInfo]` to nullable type `steps: [MWMRouterTransitStepInfo]?` and adding nil checks all over the file. Let's keep empty array for now.